instrumentation/wicket-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/wicket/WicketInstrumentationModule.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.wicket;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.tooling.InstrumentationModule;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.Arrays;
+import java.util.List;
+
+@AutoService(InstrumentationModule.class)
+public class WicketInstrumentationModule extends InstrumentationModule {
```

Comment:
Please add classloader matcher to avoid inspecting every class by type matchers.

Reply:
Is this really needed? Existing instrumentations that match only by class name don't use it unless there are separate instrumentations for different version of the same thing e.g. servlet.

Reply:
All `InstrumentationModules` should have classloader matches. Matching every class is much more expensive that eliminating the whole classloader.

Reply:
Bytebuddy applies all matchers to input class. Class loader and type matcher are run together with
```
      public boolean matches(TypeDescription typeDescription, ClassLoader classLoader, JavaModule module, Class<?> classBeingRedefined, ProtectionDomain protectionDomain) {
        return this.moduleMatcher.matches(module) && this.classLoaderMatcher.matches(classLoader) && this.typeMatcher.matches(typeDescription);
      }
```
For class loader matcher to have a positive perf effect matching class loader needs to be less expensive than matching type. If type matcher is just a `String.equals` then it is likely to be less expensive than class loader matcher which is map lookup and when not found a `ClassLoader.getResouce`. Obviously for name based matchers it would best if they were kept in map name->list of matchers which would eliminate a bunch of matching.

Reply:
Hmmm, @trask @tylerbenson @mateuszrzeszutek help me here :) I was sure that classloader matches are there for performance optimisation. Was I wrong?

Reply:
The classloader matcher was added mainly in places where the type matcher was relatively expensive (eg exploring the type hierarchy), but for a simple `named` matcher it's not really needed (unless I missed some other motivation).

Reply:
Classloader matchers defined in `TypeInstrumentation#classLoaderOptimization()` are used purely for optimization. These are usually defined when we use `extends`/`implements` type matcher.

`InstrumentationModule#classLoaderMatcher()` is used whenever there's a need to manually disable instrumenting some library versions. E.g. library had class `A` in version 1.0, but we want to instrument `[2.0,)` and the class is not directly used in instrumentation code (so muzzle does not have reference to `A`) - `classLoaderMatcher()` can be used in this case to make sure that 1.0 will never be instrumented.

Since all matchers are run together for every class I don't think it is necessary for the instrumentation module to use classloader matchers for optimization - the ones in `TypeInstrumentation` should be enough.

Reply:
Thank you for explanation! I indeed was confused :)